Wikipedia/Code/NotificationsCenterCell.swift

```diff
@@ -291,6 +301,35 @@ final class NotificationsCenterCell: UICollectionViewCell {
         updateLabels(forViewModel: viewModel)
         updateProject(forViewModel: viewModel)
         updateMetaButton(forViewModel: viewModel)
+        
+        headerLabelTapGestureRecognizer.isEnabled = isHeaderLabelTapGestureEnabled
+    }
+}
+
+//MARK: - Private
+
+private extension NotificationsCenterCell {
+    
+    var isHeaderLabelTapGestureEnabled: Bool {
```

Comment:
Nice, it looks you introduced some logical separation within `NotificationsCenterCell` here to help readability. It looks like you moved `isHeaderLabelTapGestureEnabled` here as well. I'm not suggesting you change anything here, this approach certainly works and I'm happy to merge. I just want to outline my rationale behind originally suggesting `NotificationsCenterViewModel` as its destination instead.

A property like this computed in the view model more strongly adheres to view models as something that transform more complex objects/logic into simpler "primitives" for consumption by a view, and views as something with less or only needed responsibility/knowledge of the outside world. Same vibe as how `headerText` etc. are computed in the view model now for naive consumption by the view. My approach with `NotificationsCenterCellStyle` isn't perfect - ideally, it'd be computed and originate from the view model as well, but its reliance on the active trait collection makes it a bit more direct to just treat as a view utility.

If, for example, the view model computed a boolean like `shouldAllowSecondaryTapAction` (with the same logic as `isHeaderLabelTapGestureEnabled`), the cell itself's call site could be `headerLabelTapGestureRecognizer.isEnabled = viewModel.shouldAllowSecondaryTapAction`. This moves the bulk of this logic responsibility to the view model (which has enough knowledge about the involved objects to distill the logic into a simple boolean for use by the view), reduces the responsibility of the view itself (it more naively just takes the minimum data needed to do its job of configuring itself), and as a naming bonus I think that the call site reads more naturally.

Again, I'm happy to merge this as-is, just wanted to provide my thought process if this thinking seems useful for our future decision making!

Reply:
@staykids agh, I'm sorry, I misread your original comment as asking it to be in the cell. Your explanation totally makes sense, and I like that you take any knowledge of the how out of the computed property name. I added your suggestion. Thanks as always for the careful eye 😬.

Reply:
No problem at all, thanks for the change – glad the explanation was helpful.